File: Algorithms/Sortings/Java/insertion_sort.java
import java.util.Arrays;

public class insertionSort {
	public static void main(String[] args) {
		int[] array = new int[] {5, 0, -2, 7, 3};

		for(int i = 0; i < array.length; i++) {
			int cur = array[i];
			int j;

			for(j = i; j > 0; j--) {
				if(array[j - 1] <= cur) {
					break;
				}
				array[j] = array[j - 1];
			}
			array[j] = cur;
		}
		System.out.println(Arrays.toString(array));
	}
}

File: Algorithms/Sortings/Python/insertion_sort.py
list = [43, 5, 6, 1, 0, 55, 23, -7]

for i in range(1, len(list)):
	cur = list[i]
	j = i - 1

	while j >= 0 & list[j] > cur:
		list[j + 1] = list[j]
		j -= 1
	list[j + 1] = cur